### PR TITLE
feat(sync-service): add Postgres concat(VARIADIC text) to shape subset eval

### PR DESCRIPTION
**Pull request by abegehr**
Register concat with NULL-skipping semantics (empty string when all args are NULL), matching text concat behavior distinct from || with NULL.

Tests cover literals, refs, ilike(concat(...)), contrast with ||, and document that concat() is unsupported (no concat/0 overload).

### CODE REVIEW

**Review by robacourt — CHANGES_REQUESTED** [review]
Looking good! Please could you add some tests for some of the other edge cases mentioned here: https://chatgpt.com/share/69fb490a-4c2c-8327-a6a2-74f730c95ea9 e.g. date types, variadic arrays, char(n). It's fine to just return an error when parsing so that we don't support them, but if we do parse then successfully we should handle these cases correctly.

**Comment by abegehr** [comment]
added three more tests for edge-cases

File: .changeset/moody-maps-approve.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@core/sync-service': patch
+---
+
+Add Postgres concat(variadic text) to shape subset WHERE evaluation, with NULL arguments skipped and SqlGenerator support for concat(...) round-trips.
```
Comment (robacourt) [inline .changeset/moody-maps-approve.md]:
```suggestion
Add CONCAT(variadic text) support
```

**Review by robacourt — APPROVED** [review]
Great work! Approved!
